### PR TITLE
cppcheck: 2.9.3 -> 2.10

### PR DESCRIPTION
**Pull request by rycee**
###### Description of changes

Note, the "bug hunting" feature was removed from version 2.10. It is now only available in the "premium" version of cppcheck. See

  https://github.com/danmar/cppcheck/commit/8f386e15fdedff37486c683d933ccc9a1e307388

We also add a basic installation check to avoid issues in the future.

Related #215462 and #216118

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/analysis/cppcheck/default.nix [hunk 1]
```diff
@@ -42,6 +37,16 @@ stdenv.mkDerivation rec {
     cp cppcheck.1 $man/share/man/man1/cppcheck.1
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/cppcheck/default.nix:37]:
```suggestion
    installManPage cppcheck.1
```
nit

File: pkgs/development/tools/analysis/cppcheck/default.nix [hunk 2]
```diff
@@ -42,6 +37,16 @@ stdenv.mkDerivation rec {
...
   '';
 
+  doInstallCheck = true;
+  installCheckPhase = ''
+    runHook preInstallCheck
+
+    echo 'int main() {}' > ./installcheck.cpp
+    $out/bin/cppcheck ./installcheck.cpp > /dev/null
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/cppcheck/default.nix:45]: This won't work when cross compiling, right?

Comment (rycee) [reply pkgs/development/tools/analysis/cppcheck/default.nix:45]: I believe it shouldn't be a problem, the tests shouldn't run when doing cross-compilation. See [6.5.9.1.1. doInstallCheck](https://nixos.org/manual/nixpkgs/stable/#var-stdenv-doInstallCheck):

> Cross compiled builds never run tests, no matter how doInstallCheck is set, as the newly-built program won’t run on the platform used to build it.

File: pkgs/development/tools/analysis/cppcheck/default.nix (hunk 1 above)
Comment (rycee) [reply pkgs/development/tools/analysis/cppcheck/default.nix]: Good idea, I've updated it.

**Comment by rycee** [comment]
Result of `nixpkgs-review pr 216533` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>cppcheck</li>
    <li>freeorion</li>
    <li>libsForQt5.kdev-php (plasma5Packages.kdev-php)</li>
    <li>libsForQt5.kdev-python (plasma5Packages.kdev-python)</li>
    <li>libsForQt5.kdevelop (plasma5Packages.kdevelop)</li>
    <li>libsForQt5.kdevelop-unwrapped (plasma5Packages.kdevelop-unwrapped)</li>
    <li>migraphx</li>
    <li>rsstail</li>
    <li>sqlitecpp</li>
  </ul>
</details>

**Pull request merged**